Backendless/Service/UserService.cs

```diff
@@ -18,7 +18,32 @@ public class UserService
     private static string USER_MANAGER_SERVER_ALIAS = "com.backendless.services.users.UserService";
     private ILoginStorage _loginStorage = null;
 
-    public BackendlessUser CurrentUser{ get; set; }
+    private BackendlessUser _currentUser = null;
+
+    public BackendlessUser CurrentUser
+    {
+      get
+      {
+        if( _currentUser == null )
+        {
+          if(LoginStorage.HasData)

```

Comment:
@holailusoria can "LoginStorage" be null?

Reply:
@markpiller line 29 checks for the presence of data in the LoginStorage.

Reply:
@markpiller no, it cant. An empty object is created in the LoginStorage getter if it is null. So it always returns a non-null LoginStorage object